### ISSUE TITLE
Mail clients: reporrt only the content of message body when opening an email with enter or with the mouse

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#11161]
### Steps to reproduce:
1. Open a mail client of your choice, I tried with Outlook and Thunderbird
2. Focus a mail in your mail list
3. Press enter or open it with the mouse

### Actual behavior:
NVDA reports its subject, sometimes multiple times.

### Expected behavior:
NVDA should actually start reporting  the first line of the message body since the author and the subject are already known from the navigation through the list and can also browsed in the message window as well. So no need to report this information automatically when opening the email from the list.
I don't know if this is possible, but this behavior should be restricted only to this use case. I.e. in other cases like switching between windows etc, the subject should be reported like it is already.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
All versions

#### Windows version:
all versions

#### Name and version of other software in use when reproducing the issue:
Outlook 2010, 2016, 365, Thunderbird 68.8. This might be the case in other mail clients as well.

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
all the same

#### If addons are disabled, is your problem still occuring?
yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
n/a

**Comment by Adriani90** [comment]
What I am suggsting here is kind of a standard which should apply for all mail clients. I know that NVDA will not be abled to achieve this for every client, but at least for Outlook and Thunderbird this would be quite nice.

**Comment by Dennisl123** [comment]
My issue shouldn't have been closed.  I am asking for an option to have it automatically read the first line or not.  This is not currently available.

**Comment by josephsl** [comment]
Hi,

If I understand this thread (spanning over at least three issues) correctly, you want NVDA to not read first line of a document automatically, or at least let users decide this behavior, correct? If yes, this requires consideration about focus change behavior, as this is a result of a document gaining focus. Here's what happens:

1. You open a document (an email message, for example).
2. Header information is announced.
3. Windows changes the system focus so the cursor is now located at the start of the message.
4. Focus change is noticed by accessibility API's (IAccessible/MSAA, in this case) and focus change event is sent to NVDA and others.
5. NVDA will react to focus change event (specifically, gain focus/navigator object events) by locating where the system cursor is and announcing the line where it is located (this involves probing what's known as "text info", a component used to identify text that is part of a control)).
7. Speech and/or braille will present the line where the cursor is located.

What Dennis is asking is configuring what happens by the time NVDA moves to steps 6 and 7. The bright side of this change is that users may feel NVDA is more efficient. The downside of this change is loss of cursor announcement, needed to find out where one is locatet, especially when focus moves to a document.

Although I understand the efficiency argument, I'm a bit hesitant to recommend the new behavior due to user experience consistency: NVDA is supposed to react to focus changes if it decides that it is okay to do so. The reason for announcing the first line of a document is due to focus change and helping users learn what they are dealing with, and in case of text fields and documents, where they are located. Efficiency is important for sake of productivity. But we cannot ignore why a screen reader exists: to help users know where they are so they can be more productive for a given situation.

Note that I'm not speaking for NV Access because I don't work for them - I'm presenting personal views here. I advise leaving this issue open until folks can come to an agreement that:

* This is something we should look into.
* The proposal is not workable.

Thanks.

**Comment by Adriani90** [comment]
No, just ignore the other three issues there was a miss understanding. Actually it came to my mind independnetly, it would reduce verbosity if we do not hear the subject of the email when pressing enter on it from the message list.

 

 

Von: Joseph Lee <notifications@github.com> 
Gesendet: Mittwoch, 13. Mai 2020 22:55
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Mail clients: reporrt only the content of message body when opening an email with enter or with the mouse (#11161)

 

Hi,

If I understand this thread (spanning over at least three issues) correctly, you want NVDA to not read first line of a document automatically, or at least let users decide this behavior, correct? If yes, this requires consideration about focus change behavior, as this is a result of a document gaining focus. Here's what happens:

1.	You open a document (an email message, for example).
2.	Header information is announced.
3.	Windows changes the system focus so the cursor is now located at the start of the message.
4.	Focus change is noticed by accessibility API's (IAccessible/MSAA, in this case) and focus change event is sent to NVDA and others.
5.	NVDA will react to focus change event (specifically, gain focus/navigator object events) by locating where the system cursor is and announcing the line where it is located (this involves probing what's known as "text info", a component used to identify text that is part of a control)).
6.	Speech and/or braille will present the line where the cursor is located.

What Dennis is asking is configuring what happens by the time NVDA moves to steps 6 and 7. The bright side of this change is that users may feel NVDA is more efficient. The downside of this change is loss of cursor announcement, needed to find out where one is locatet, especially when focus moves to a document.

Although I understand the efficiency argument, I'm a bit hesitant to recommend the new behavior due to user experience consistency: NVDA is supposed to react to focus changes if it decides that it is okay to do so. The reason for announcing the first line of a document is due to focus change and helping users learn what they are dealing with, and in case of text fields and documents, where they are located. Efficiency is important for sake of productivity. But we cannot ignore why a screen reader exists: to help users know where they are so they can be more productive for a given situation.

Note that I'm not speaking for NV Access because I don't work for them - I'm presenting personal views here. I advise leaving this issue open until folks can come to an agreement that:

*	This is something we should look into.
*	The proposal is not workable.

Thanks.

**Comment by Dennisl123** [comment]
Yes Joseph That is what I’m wanting to give the User the choice how it behaves.  It can behave as it does by default but add options for the user to control its behavior.

**Comment by Adriani90** [comment]
Giving the user the control of every really small impact option means to overload a program with settings at the cost of novice users. Anyway, this issue seems  not related, so please keep discussions on this topic in the other issue.

 

 

Von: Dennisl123 <notifications@github.com> 
Gesendet: Mittwoch, 13. Mai 2020 22:59
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Mail clients: reporrt only the content of message body when opening an email with enter or with the mouse (#11161)

 

Yes Joseph That is what I’m wanting to give the User the choice how it behaves. It can behave as it does by default but add options for the user to control its behavior.

**Comment by josephsl** [comment]
Hi, can you answer the follow-up question: can you think of other impact besides efficiency? This is so that people wishing to implement this can better understand what to do. Thanks.

**Comment by Adriani90** [comment]
What do you mean? Most impact that new iissues or features have are related to efficiency.

Well, it might also impact performance positively if NVDA does not fetch the subject when opening the email with enter which could lead to the ne email opening faster.

 

 

Von: Joseph Lee <notifications@github.com> 
Gesendet: Mittwoch, 13. Mai 2020 23:03
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Mail clients: reporrt only the content of message body when opening an email with enter or with the mouse (#11161)

 

Hi, can you answer the follow-up question: can you think of other impact besides efficiency? This is so that people wishing to implement this can better understand what to do. Thanks.

**Comment by josephsl** [comment]
Hi, message content and headers/subject line are two different things (subject line is read because it is the title of the document one is reading). My follow-up question deals with experience for users who may enable the proposed option in other places, not necessarily performance (yet), because looking at performance requires examining factors such as memory usage, response from API DLL’s, program in question, Python, and others. Thanks.

**Comment by Adriani90** [comment]
yes, that's why I wanted to keep this issue splitted from the issues targeting the message body after noticing the missunderstanding. Discussions on message body are on #3020 However, what Denis requests can be achieven by pressing ctrl key right after pressing enter. In my view no need for yet an additional setting with low impact here.
Otherwise, we need more justification what this would bring to an user or what that setting is exactly intended to improve.

**Comment by Dennisl123** [comment]
User control some people prefer it to not speak what is read automatically.  Some people read line by line.   Also other screen readers offer the option to turn off automatic reading of the body of the email.  If you aren’t wanting to make as part of NVDA core could an add-on be made to accomplish this?  I would have no idea how to do this.

**Comment by josephsl** [comment]
Hi, I think we are driving a parallel road: what I’m actually asking is to consider why users choose to ask screen readers to not read the line where the cursor is located when opened (top line). Although we do need to consider user preference, without understanding why, it may give rise to code that really doesn’t solve the problem users are encountering. Thanks.

**Comment by Adriani90** [comment]
The simplest way would be if Denis creates a new feature request describing exactly the expectations and justifying why users need this, keeping in mind the template. This issue here is not related to Denis‘s request. See issue description. Thanks.


Von meinem iPhone gesendet

**Comment by josephsl** [comment]
Hi,

The actual issue that Dennis is refering to is #11157. Let's discuss that aspect further on that issue.

Thanks.

**Comment by XLTechie** [comment]
Please correct the spelling of "report" in the title of this issue. And feel 
free to delete this message. :)

**Comment by Brian1Gaff** [comment]
I'm happy with it as it is, so this should be on a selection system . I 
still use Outlook express and this is old app module, if such messing with 
focus occurs it could potentially affect the way older  mail and newsreaders 
work, including older versions of Tbird, hence the idea of aa toggle. One 
thing ideally I'd like to see in newsgroups readers is the ability to hear 
the groups in the list you are responding to, without having to tab about to 
find them, as trolls often add totally inappropriate groups to cause flame 
wars.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users

**Comment by Brian1Gaff** [comment]
It might, but might not. There are so many different ways email software 
presents the various to from subject etc, lines depending on the client in 
use. It might be more likely to be able to make add ons for a particular 
client if in use rather than attempting to make a one size fits all solution 
which could spoil how it works now by some, as yet unrealised change.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users

**Comment by Adriani90** [comment]
@Brian1Gaff excluding the subject from reporting when opening the email with enter or with the mouse would not have any negative impact on the focus tracking. It is actually just an additional object rendered which is not related to the system carret. So your concerns are not related here.

**Comment by Novalis7747** [comment]
Adriani, from my point of view NVDA is not repeating the subject line when opening a mail in Outlook. It just reads the title of the window as a normal behavior when a window opens. After reading the window title, which is almost identical of the subject line plus content type, it is positioning in the message body and reads the first line automatically, but only if there is text in the first line of course. So your ask would be to suppress the reading of the window title when opening a mail. I would support this ask, because as you already stated, this is redundant information and time consuming. 
The reading of the first line of the message content needs no correction at my end coz it behaves as designed here.

cheers,
  Ralf

**Comment by Dennisl123** [comment]
Hi, just what the status of this is?  I'm hoping it is added to NVDA soon.
